platform/platform-filter-clientinfo/src/main/java/org/codice/ddf/platform/filter/clientinfo/impl/ClientInfoFilter.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.filter.clientinfo.impl;
+
+import static org.codice.ddf.platform.filter.clientinfo.constants.ClientInfoKeys.CLIENT_INFO_KEY;
+import static org.codice.ddf.platform.filter.clientinfo.constants.ClientInfoKeys.CONTEXT_PATH_KEY;
+import static org.codice.ddf.platform.filter.clientinfo.constants.ClientInfoKeys.REMOTE_ADDRESS_KEY;
+import static org.codice.ddf.platform.filter.clientinfo.constants.ClientInfoKeys.REMOTE_HOST_KEY;
+import static org.codice.ddf.platform.filter.clientinfo.constants.ClientInfoKeys.SCHEME_KEY;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.apache.shiro.util.ThreadContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This CXF filter extracts client-specific information and places it in the shiro {@link ThreadContext}
+ * so it can be forwarded to the {@code CatalogFramework} using a {@code PrePolicyPlugin}.
```

Comment:
Don't think we should refer to the `CatalogFramework` or a specific plugin here. Anybody, not only plugins, could actually use those values.